common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -7247,7 +7247,7 @@ public List<Map.Entry<String, String>> getMatchingEntries(Pattern regex) {
     for (Map.Entry<String, String> entry : this) {
```

Comment:
that is some crazy code, could you please refactor it with below snippet and extract to HS2 (where it's used)
````
  public Map<String, Integer> getPoolConfigEntries(HiveConf hiveConf) {
    Map<String, Integer> poolConfigs = new HashMap<>();
    for (Map.Entry<String, String> entry : hiveConf) {
      Matcher matcher = COMPACTION_POOLS_PATTERN.matcher(entry.getKey());
      if (matcher.matches()) {
        matchingEntries.put(matcher.group(1), NumberUtils.toInt(entry.getValue(), 0));
      }
    }
    return poolConfigs;
  }
````
wonder how it was even tested?

Reply:
Done